app/scripts/modules/core/src/instance/InstanceListBody.tsx

```diff
@@ -0,0 +1,249 @@
+import * as React from 'react';
+import { BindAll } from 'lodash-decorators';
+import { Subject } from 'rxjs';
+import * as classNames from 'classnames';
+
+import { IServerGroup, IInstance, ILoadBalancerHealth } from 'core/domain';
+import { ReactInjector } from 'core/reactShims';
+import { timestamp } from 'core/utils/timeFormatters';
+import { Tooltip } from 'core/presentation';
+
+export interface IInstanceListBodyProps {
+  serverGroup: IServerGroup;
+  instances: IInstance[];
+  hasDiscovery: boolean;
+  hasLoadBalancers: boolean;
+}
+
+export interface IInstanceListBodyState {
+  selectedInstanceIds: string[];
+  activeInstanceId: string;
+  multiselect: boolean;
+}
+
+@BindAll()
+export class InstanceListBody extends React.Component<IInstanceListBodyProps, IInstanceListBodyState> {
+
+  private clusterFilterModel = ReactInjector.clusterFilterModel.asFilterModel;
+  private MultiselectModel = ReactInjector.MultiselectModel;
+  private $uiRouter = ReactInjector.$uiRouter;
+  private $state = ReactInjector.$state;
+  private destroy$ = new Subject();
+
+  constructor(props: IInstanceListBodyProps) {
+    super(props);
+    this.MultiselectModel.instancesStream
+      .takeUntil(this.destroy$)
+      .subscribe(() => {
```

Comment:
There was some pattern about subscribing in `componentDidMount` but I don't remember why.

Reply:
no sense in asking why

Reply:
I wish I could find the article I was reading the other day... but alas.

The React core team is currently discouraging people from adding side effects in constructors/`componentWillMount` callbacks in anticipation of async rendering, because construction is no longer going to be a guarantee that your component is actually getting mounted/rendered. [This comment](https://github.com/reactjs/reactjs.org/issues/302#issuecomment-345445888) is the closest I could find to the actual post I read about it :/

Reply:
That makes sense. Also interesting to see `componentWillMount` might be called multiple times. (I don't think we use it much, if at all, but good to keep in mind)